src/test/server/store-test.js

```diff
@@ -1,9 +1,203 @@
-const UserStore = require('../../../_compiled_/stores/user-store').default;
+const assert = require('assert');
+
+const usersData  = require('../../test/data/users.json');
+const UserAction = require('../../../_compiled_/actions/user-action').default;
+const UserInfo   = require('../../../_compiled_/stores/user-info').default;
+const UserStore  = require('../../../_compiled_/stores/user-store').default;
+const EVENT_TYPE = require('../../../_compiled_/stores/event-type').default;
+
+const TEST_USER = new UserInfo(
+    {
+        name:    'Ervin New',
+        phone:   '89500418181',
+        email:   'vika.chernookaya@mail.ru',
+        website: 'github.com'
+    }
+);
+
+const TEST_USER_EXISTING = new UserInfo(
+    {
+        name:    'Ervin Howell',
+        phone:   '0418181',
+        email:   'ervin@mail.ru',
+        website: 'github.com'
+    }
+);
+
+const SEARCH_USER_INFO = new UserInfo({ name: 'Ervin' });
+
+const SEARCH_NON_EXIST_USER_INFO = new UserInfo({ name: 'Ivan' });
+
+const UPDATE_USER = new UserInfo(
+    {
+        id:      2,
+        name:    'Vika',
+        phone:   '555',
+        email:   'ervin@mail.ru',
+        website: 'github.com'
+    }
+);
+
+const SEARCH_FIELDS = {
+    name:    true,
+    phone:   false,
+    email:   false,
+    website: false
+};
+
+function subEvent (name, raiseAction) {
+    return new Promise(resolve => {
+        UserStore.on(name, (...args) => {
+            resolve(args);
+        });
+        raiseAction();
+    });
+}
+
+function _getInitialUserList () {
+    return usersData.map((user) => {
+        return new UserInfo(
+            {
+                id:      user.id,
+                name:    user.name,
+                phone:   user.phone,
+                email:   user.email,
+                website: user.website
+            },
+            user.address,
+            user.company
+        );
+    });
+}
 
 describe('UserStore', () => {
+    afterEach(() => {
+        UserStore._users        = _getInitialUserList();
+        UserStore._searchedInfo = null;
+        UserStore.removeAllListeners(EVENT_TYPE.userAdded);
```

Comment:
Can we call ` UserStore.removeAllListeners()` without arguments to unsubscribe all listeners for all events at the same time?